### ISSUE TITLE
Bug: commands do not work when Obsidian is already opened

### ISSUE THREAD

**Issue by mora-unie-youer** [chrisgrieser/shimmering-obsidian#191]
### Make sure you have done the following

- [X] I have [updated to the latest version](https://github.com/chrisgrieser/shimmering-obsidian/releases/latest) of this workflow.
- [X] I have read the documentation of the feature I am having problems with.
- [X] I am using Alfred 5. (Alfred 4 is not supported by this workflow anymore.)
- [X] I have updated to the latest version of the Metadata-Extractor and the Advanced URI plugin.
- [X] If the `o…` main search does not work, I made sure that I haven't changed any of the default settings of the Metadata-Extractor. Furthermore, I have tried rerunning `osetup` and it still does not work.
- [X] If notes in general can be found, but recently created or renamed notes cannot, I used the `oupdate` command to manually update the metadata.

### Bug Description

When I try to open note by using `o` with already opened Obsidian, workflow will do nothing. If Obsidian is closed, workflow will successfully open requested note. (so there's no issues with Obsidian addons)

I'm new macOS user (but hugely experienced Linux user), so I **understand** that the issue is not directly in workflow, but in some kind of permissions. But attempts to google it failed, so I try to find help here. In relevant screenshots I attached screenshots of all permissions that Alfred seems to require.

### Relevant Screenshot

Granted permissions for Alfred (some of them required me to add application from Applications folder).
<img width="827" alt="image" src="https://github.com/user-attachments/assets/8a7b512d-60a3-412d-a4a2-4efac7191acd">
<img width="827" alt="image" src="https://github.com/user-attachments/assets/149876b3-015b-480e-bce1-c1a0a27c0b74">
<img width="827" alt="image" src="https://github.com/user-attachments/assets/4076e598-cfa9-4c33-9f9f-b2797987e90a">
<img width="827" alt="image" src="https://github.com/user-attachments/assets/6e2c250d-4521-4c58-8e39-1e5e8252a57d">

### To Reproduce

1. Use `o <part of title>` to open note
  a) If Obsidian is already opened, error from debugging log will be thrown
  b) If Obsidian is closed, it will open with required note opened

### Debugging Log

```Text
[12:45:37.023] Logging Started...
[12:45:40.238] ERROR: Shimmering Obsidian[Run Script] /Users/mora/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.D336BE40-93A8-4D0E-A73E-3B02F3954E0C/scripts/open-note.js: execution error: Error: Error: osascript is not allowed assistive access. (-25211)
```

### Workflow Configuration

<img width="1292" alt="image" src="https://github.com/user-attachments/assets/2b9055a5-f5f9-472c-9152-ab7f35451b90">
<img width="1292" alt="image" src="https://github.com/user-attachments/assets/aebd32ce-a33f-44ba-afde-f147e5275b04">
<img width="1292" alt="image" src="https://github.com/user-attachments/assets/28534940-5d3b-4720-bfba-381d34f1897b">
<img width="1292" alt="image" src="https://github.com/user-attachments/assets/fdc008b4-a140-48fb-b754-75b05bfb2320">

### Are you using Alfred in Compatibility Mode?

- [ ] I am using Compatibility Mode.

### Version & System Information

```Text
-------------------------------
INTERNAL WORKFLOW CONFIGURATION
Vault Path: /Users/mora/Mynd
config folder: exists
Metadata JSONs: 4/4
metadata.json String Length: 147423
-------------------------------
METADATA EXTRACTOR CONFIG
{"writingFrequency":"30","writeFilesOnLaunch":true}
-------------------------------
WORKSPACE DATA
'workspace.json' exists
-------------------------------
SYSTEM
macOS: 15.0
Alfred: 5.5
-------------------------------
INSTALLED VERSION
Obsidian: .asar file missing
This Workflow: 4.9.5
Advanced URI Plugin: 1.44.1
Metadata Extractor: 1.1.0
-------------------------------
LATEST VERSION
Obsidian: 1.6.7 (Insider: 1.7.3)
This Workflow: 4.9.5
Advanced URI Plugin: 1.44.1
Metadata Extractor: 1.1.0
-------------------------------
```

**Comment by chrisgrieser** [comment]
You are right, `osascript is not allowed assistive access. (-25211)` indicates a permission issue, [in particular this line failing](https://github.com/chrisgrieser/shimmering-obsidian/blob/ed0c50228c44ca5a93161928354b81a02dd5258b/scripts/open-note.js#L8), it seems.

I gave it a try, and on my it's the permission `System Settings → Privacy & Security → Accessibility → Alfred 5`. Disabling this one causes the same error for me, enabling it makes the error go away. However, on one of your screenshot it looks like you have that permission already enabled, so I am a bit at a loss here. Maybe try disabling and then re-enabling that permission, and then restarting Alfred or something like that in case there is an macOS issue with the permission? 

Seeing you are on macOS 15.0, it could be that the system update somehow messed with the permissions. I am still on macOS 14, so it could be that the permission somehow moved location.

---

If that does not help, you'd unfortunately have to debug it on your end, since my ability to help with system level issues is somewhat limited. I created a minimal debugging-workflow. Download it, unzip, and install the workflow:

[Debug.zip](https://github.com/user-attachments/files/17174432/Debug.zip)

It does nothing other than running the problematic line via Alfred, and is in my case dependent on the assistive access for osascript. Open the console for that workflow, and trigger the line via `debug` as Alfred keyword. The same error should occur then. This might help you in experimenting things on your end.

Unfortunately, this is as much as I help do on my end, since it is an issue between Alfred and macOS, and otherwise not related to the workflow, sorry. Keep me posted if you find out how to fix the permission issue.
